### PR TITLE
Add Email Crud Operations using Az Dev Tools

### PR DESCRIPTION
**Pull request by ddouglas-msft**
Adds operations for the Azure Communication Services Email Service
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by kairu-ms — APPROVED** [review]

File: src/communication/azext_communication/version.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # --------------------------------------------------------------------------------------------
 
 
-VERSION = '1.8.0b1'
+VERSION = '1.8.1'
```
Comment (kairu-ms) [inline src/communication/azext_communication/version.py:7]: Please update the release history in this [file](https://github.com/Azure/azure-cli-extensions/blob/main/src/communication/HISTORY.rst)

Comment (ddouglas-msft) [reply src/communication/azext_communication/version.py:7]: Thank you. I've updated that file.

File: src/communication/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 Release History
 ===============
+1.8.1
```
Comment (anmolbohra97) [inline src/communication/HISTORY.rst:5]: Update this version to 1.8.1b1

Comment (ddouglas-msft) [reply src/communication/HISTORY.rst]: Thanks @anmolbohra97. I've updated the version in
        modified:   src/communication/HISTORY.rst
        modified:   src/communication/azext_communication/version.py

File: src/communication/azext_communication/aaz/latest/communication/email/__cmd_group.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
```
Comment (AllyW) [inline src/communication/azext_communication/aaz/latest/communication/email/__cmd_group.py:12]: Please upload your local aaz repo change and make a pull request in aaz repo.

**Comment by AllyW** [comment]
@ddouglas-msft please also make a pr in aaz repo with the code changes in dev env

File: src/communication/azext_communication/version.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 # --------------------------------------------------------------------------------------------
 
 
-VERSION = '1.8.0b1'
+VERSION = '1.8.1b1'
```
Comment (jsntcy) [inline src/communication/azext_communication/version.py:7]: Based on our [CLI Extension Version Transition Table](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/versioning_guidelines.md#cli-extension-version-transition-table), the version should be `1.8.0b2`. 
Please also update `azext.isPreview` to `true` in azext_metadata.json as well if you want to release as preview version.

**Comment by ddouglas-msft** [comment]
> @ddouglas-msft please also make a pr in aaz repo (https://github.com/Azure/aaz) with the code changes in dev env

The updates are in this PR.

https://github.com/Azure/aaz/pull/408

**Review by kagbakpem — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by ddouglas-msft** [comment]
> Please fix CI issues

Thanks. I have fixed the issue with the indent.

**Comment by ddouglas-msft** [comment]
Hello @yonzhan and @AllyW  . Is there anything else missing?

**Comment by ddouglas-msft** [comment]
Hello @AllyW, @yonzhan, @kairu-ms, @jsntcy. Any updates on this PR?

File: src/communication/azext_communication/azext_metadata.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {
-    "azext.minCliCoreVersion": "2.49.0"
+    "azext.minCliCoreVersion": "2.55.0"
```
Comment (jsntcy) [inline src/communication/azext_communication/azext_metadata.json:2]:
```suggestion
    "azext.minCliCoreVersion": "2.55.0",
    "azext.isPreview": true
```

Comment (jsntcy) [reply src/communication/azext_communication/azext_metadata.json:2]: As the extension is in preview (1.8.0b2), so you should set `azext.isPreview` to `true`

File: src/communication/HISTORY.rst (hunk 2 above)
Comment (ddouglas-msft) [reply src/communication/HISTORY.rst]: Updated to 1.8.0b2 per comment from @jsntcy

File: src/communication/azext_communication/aaz/latest/communication/email/__cmd_group.py (hunk 3 above)
Comment (ddouglas-msft) [reply src/communication/azext_communication/aaz/latest/communication/email/__cmd_group.py:12]: Thank you. https://github.com/Azure/aaz/pull/408

File: src/communication/azext_communication/version.py (hunk 4 above)
Comment (ddouglas-msft) [reply src/communication/azext_communication/version.py]: This has been updated per your other PR feedback

**Review by necusjz — APPROVED** [review]

**Pull request merged**